test/e2e/user_workload_monitoring_test.go

```diff
@@ -531,9 +533,20 @@ func assertTenancyForMetrics(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// Grant enough permissions to the account so it can read metrics.
+	// Grant just-enough permissions to the account so it can read metrics.
 	err = framework.Poll(2*time.Second, 10*time.Second, func() error {
-		_, err = f.CreateRoleBindingFromClusterRole(userWorkloadTestNs, testAccount, "admin")
+		_, err = f.CreateRoleBindingFromTypedRole(userWorkloadTestNs, testAccount, &rbacv1.Role{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "tenancy-test-metrics",
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					APIGroups: []string{"metrics.k8s.io"},
+					Resources: []string{"pods"},
+					Verbs:     []string{"get"},
```

Comment:
we can also add new test scenarios like
* POST queries denied for roles without 'create' verb.
* POST queries allowed for roles with 'create' verb.